app/qml/components/PasswordField.qml

```diff
@@ -0,0 +1,95 @@
+import QtQuick 2.7
+import QtQuick.Controls 2.0
+import QtQuick.Layouts 1.3
+import QtGraphicalEffects 1.0
+import lc 1.0
+import "../" // import InputStyle singleton
+
+Row {
+  id: root
+  spacing: 0
+
+  property var fontColor
+  property var bgColor
+  property var password: password
+  property string placeholder: 'Password'
+
+  Rectangle {
+    id: iconContainer2
+    height: root.height
+    width: root.height
+    color: root.bgColor
+
+    Image {
+      anchors.margins: (root.height / 4)
+      id: icon
+      height: root.height
+      width: root.height
+      anchors.fill: parent
+      source: '../lock.svg'
+      sourceSize.width: width
+      sourceSize.height: height
+      fillMode: Image.PreserveAspectFit
+    }
+
+    ColorOverlay {
+      anchors.fill: icon
+      source: icon
+      color: root.fontColor
+    }
+  }
+
+  TextField {
+    id: password
+    width: loginForm.width - iconContainer.width - visibilityIconContainer.width
+    height: fieldHeight
+    font.pixelSize: InputStyle.fontPixelSizeNormal
+    color: root.fontColor
+    placeholderText: qsTr(root.placeholder)
```

Comment:
Here I think qsTr does not work - it needs a constant string otherwise the tool that extracts translatable strings can't extract it